scmgr/app.go

```diff
@@ -0,0 +1,562 @@
+	}
+	group := readGroup(c, 0)
+	client := skipchain.NewClient()
+	hash, err := hex.DecodeString(c.Args().Get(1))
+	if err != nil {
+		return err
+	}
+	gcr, cerr := client.GetUpdateChain(group.Roster, hash)
+	if cerr != nil {
+		return cerr
+	}
+	latest := gcr.Update[len(gcr.Update)-1]
+	genesis := latest.GenesisID
+	if genesis == nil {
+		genesis = latest.Hash
+	}
+	log.Infof("Joined skipchain %x", genesis)
+	cfg := getConfigOrFail(c)
+	cfg.Sbm.Store(latest)
+	log.ErrFatal(cfg.save(c))
+	return nil
+}
+
+// Returns the number of calls.
+func add(c *cli.Context) error {
+	log.Info("Adding a block with a new group")
+	if c.NArg() < 2 {
+		return errors.New("Please give group-file and id to add")
+	}
+	group := readGroup(c, 1)
+	cfg := getConfigOrFail(c)
+	sb := cfg.Sbm.GetFuzzy(c.Args().First())
+	if sb == nil {
+		return errors.New("didn't find latest block - update first")
+	}
+	client := skipchain.NewClient()
+	guc, cerr := client.GetUpdateChain(sb.Roster, sb.Hash)
+	if cerr != nil {
+		return cerr
+	}
+	latest := guc.Update[len(guc.Update)-1]
+	ssbr, cerr := client.StoreSkipBlock(latest, group.Roster, nil)
+	if cerr != nil {
+		return errors.New("while storing block: " + cerr.Error())
+	}
+	cfg.Sbm.Store(ssbr.Latest)
+	log.ErrFatal(cfg.save(c))
+	log.Infof("Added new block %x to chain %x", ssbr.Latest.Hash, ssbr.Latest.GenesisID)
+	return nil
+}
+
+// Adds a block with the page inside.
+func addWeb(c *cli.Context) error {
+	log.Info("Adding a block with a page")
+	if c.NArg() < 2 {
+		log.Fatal("Please give skipchain-id and html-file to save")
+	}
+	for i, s := range c.Args() {
+		log.Print(i, s)
+	}
+	cfg := getConfigOrFail(c)
+	sb := cfg.Sbm.GetFuzzy(c.Args().First())
+	if sb == nil {
+		return errors.New("didn't find latest block - update first")
+	}
+	client := skipchain.NewClient()
+	guc, cerr := client.GetUpdateChain(sb.Roster, sb.Hash)
+	if cerr != nil {
+		return cerr
+	}
+	latest := guc.Update[len(guc.Update)-1]
+	log.Print("Reading file", c.Args().Get(1))
+	data, err := ioutil.ReadFile(c.Args().Get(1))
+	log.ErrFatal(err)
+	ssbr, cerr := client.StoreSkipBlock(latest, nil, &html{data})
+	if cerr != nil {
+		return errors.New("while storing block: " + cerr.Error())
+	}
+	cfg.Sbm.Store(ssbr.Latest)
+	log.ErrFatal(cfg.save(c))
+	log.Infof("Added new block %x to chain %x", ssbr.Latest.Hash, ssbr.Latest.GenesisID)
+	return nil
+}
+
+// Updates a block to the latest block
+func update(c *cli.Context) error {
+	log.Info("Updating block")
+	if c.NArg() < 1 {
+		return errors.New("please give block-id to update")
+	}
+	cfg := getConfigOrFail(c)
+
+	sb := cfg.Sbm.GetFuzzy(c.Args().First())
+	if sb == nil {
+		return errors.New("didn't find latest block in local store")
+	}
+	client := skipchain.NewClient()
+	guc, cerr := client.GetUpdateChain(sb.Roster, sb.Hash)
+	if cerr != nil {
+		return errors.New("while updating chain: " + cerr.Error())
+	}
+	if len(guc.Update) == 1 {
+		log.Info("No new block available")
```

Comment:
yes, but it might be useful to have the genesis-id, and hash of the last block anyway.